src/lib/Monads.ml

```diff
@@ -131,48 +132,34 @@ struct
   let abort_if_inconsistent : 'a -> 'a m -> 'a m =
     fun abort m ->
     fun st ->
-    match CofEnv.status st.cof_env with
+    match CofEnv.already_inconsistent st.cof_env with
```

Comment:
this is breaking an important invariant. this function is not an optimization — it is important that we definitely abort when it is inconsistent (not just sometimes), because we may hit “type errors” during computation that are actually OK because of inconsistent context.

Reply:
I'm confused. Is the previous `CofEnv.status` precise? I believe I did not change the semantics. I can write an expensive checking for this.

Reply:
the status was getting updated in the old code of “assume”. maybe there was a bug (?) but the intent was to be precise and not approximate.

Reply:
Oh I see. You are right. Let me fix this.

Reply:
Actually the code is already correct LOL it's just that my comment is wrong.